storage/memory/stats_buffer.go

```diff
@@ -43,7 +46,65 @@ func (self *StatsBuffer) Add(item *info.ContainerStats) {
 	self.buffer[self.index] = *item
 }
 
+// Returns up to maxResult elements in the specified time period (inclusive).
+// Results are from first to last. maxResults of -1 means no limit.
+func (self *StatsBuffer) InTimeRange(start, end time.Time, maxResults int) []*info.ContainerStats {
+
+	// NOTE: Since we store the elments in descending timestamp order "start" will
+	// be a higher index than "end".
+
+	var startIndex int
+	if start.IsZero() {
+		// None specified, start at the beginning.
+		startIndex = self.size - 1
```

Comment:
Only if the size is 0. It does get handled it seems (although by no design of mine :P). When size == 0, start = -1 and end = 0, which means numResults = 0.

Open to suggestions on making this behavior more clear :)